### ISSUE TITLE
Pagination cutoff at both 400% zoom and 200% text scale

### ISSUE THREAD

**Issue by ravenokko** [shopware/platform#18724]
### Shopware Version

trunk

### Affected area / extension

Platform(Default)

### Actual behaviour

Pagination is cut off at 320px width (skip to end double arrow icon off screen). This does not meet the requirements for [WCAG S.C. 1.4.10](https://www.w3.org/WAI/WCAG22/Understanding/reflow.html). 

<img width="25%" height="25%" alt="Image" src="https://github.com/user-attachments/assets/f65cfafd-b9f8-4efb-8b57-c0138bc1dd4d" />

Pagination is also cut off at 320px width with text zoomed to 200%. This does not meet the requirements for [WCAG S.C. 1.4.4](https://www.w3.org/WAI/WCAG22/Understanding/resize-text.html). 

<img width="25%" height="25%" alt="Image" src="https://github.com/user-attachments/assets/3458670f-8da0-4668-af5c-28ddf6254dfa" />

### Expected behaviour

Pagination numbers and icons adapt to screen and text zoom dimensions. See Tabular data and grid-based UI in [WCAG S.C. 1.4.10](https://www.w3.org/WAI/WCAG22/Understanding/reflow.html)

### How to reproduce

- In Firefox go to Preferences > General > Zoom
   - Ensure default Zoom is set to 100%
   - check Zoom text only
- Create shop with enough products to trigger pagination
- Visit storefront homepage with pagination
   - Set viewport to 320px wide (Test 1) 
   - Zoom to 200% (Test 2)
